sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -103,8 +119,26 @@ protected String visitIsNotNull(String v) {
     return v + " IS NOT NULL";
   }
 
+  protected String visitStringPredicate(String name, String l, String r) {
+    String value = r.replaceAll("'", "");
+    switch (name) {
+      case "STARTS_WITH":
+        return l + " LIKE '" + value + "%'";
+      case "ENDS_WITH":
+        return l + " LIKE '%" + value + "'";
+      default: // "CONTAINS"
+        return l + " LIKE '%" + value + "%'";
+    }
+  }
+
   protected String visitBinaryComparison(String name, String l, String r) {
-    return "(" + l + ") " + name + " (" + r + ")";
+    switch (name) {
+      case "<=>":
+        return "(NOT (" + l + " != " + r + " OR " + l + " IS NULL OR " + r + " IS NULL) OR " +
+          "(" + l + " IS NULL AND " + r + " IS NULL))";
```

Comment:
shouldn't this be `(a = b) or (a is null and b is null)`?

Reply:
I guess `EqualNullSafe` means compare with null for safety.

Reply:
doesn't `(a = b) or (a is null and b is null)` return the same result as `a <=> b`?

Reply:
OK. I got it.